compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/npu_dma_to_half_dma_cpy_nd.mlir

```diff
@@ -0,0 +1,240 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(iree-amdaie-npu-dma-to-half-dma-cpy-nd)" --split-input-file --verify-diagnostics %s | FileCheck %s
+
+// CHECK-LABEL: @no_amdaie_device
+module {
+  func.func @no_amdaie_device() {
+    amdaie.workgroup {
+      amdaie.controlcode {
+        amdaie.end
+      }
+    }
+    return
+  }
+}
+
+// -----
+
+// CHECK-LABEL: @no_ops
+#executable_target_amdaie_xclbin_fb = #hal.executable.target<"amd-aie", "amdaie-xclbin-fb", {target_device = "npu1_4col", ukernels = "none"}>
+module attributes {hal.executable.target = #executable_target_amdaie_xclbin_fb} {
+  func.func @no_ops() {
+    amdaie.workgroup {
+      amdaie.controlcode {
+        amdaie.end
+      }
+    }
+    return
+  }
+}
+
+// -----
+
+// CHECK-LABEL: @npu_dma_cpy_nd_source
+// CHECK: %[[BD_ID:.+]] = amdaie.bd_id
+// CHECK: %[[OBJECT_FIFO_0:.+]] = amdaie.logicalobjectfifo.from_buffers
+// CHECK: %[[CHANNEL_0:.+]] = amdaie.channel
+// CHECK: %[[CHANNEL_1:.+]] = amdaie.channel
+// CHECK: %[[CONNECTION:.+]] = amdaie.connection
+// CHECK: %[[OBJECT_FIFO_1:.+]] = amdaie.logicalobjectfifo.from_memref
+// CHECK: amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_1]] [] [] [] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
+// CHECK: amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
+// CHECK: %[[TOKEN_0:.+]] = amdaie.npu.half_dma_cpy_nd async %[[CONNECTION]](%[[OBJECT_FIFO_1]] [0, 0, 0, 0] [2, 4, 16, 16] [0, 64, 8, 1] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
+// CHECK: amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
+// CHECK: amdaie.npu.dma_wait(%[[TOKEN_0]] : !amdaie.async_token)
+// CHECK: %[[TOKEN_1:.+]] = amdaie.npu.half_dma_cpy_nd async %[[CONNECTION]](%[[OBJECT_FIFO_1]] [0, 0, 0, 32] [2, 4, 16, 16] [128, 64, 8, 1] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
+// CHECK: amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
+// CHECK: amdaie.npu.dma_wait(%[[TOKEN_1]] : !amdaie.async_token)
```

Comment:
```suggestion
// CHECK-LABEL: @npu_dma_cpy_nd_source
// CHECK:             %[[BD_ID:.+]] = amdaie.bd_id
// CHECK:             %[[OBJECT_FIFO_0:.+]] = amdaie.logicalobjectfifo.from_buffers
// CHECK:             %[[CHANNEL_0:.+]] = amdaie.channel
// CHECK:             %[[CHANNEL_1:.+]] = amdaie.channel
// CHECK:             %[[CONNECTION:.+]] = amdaie.connection
// CHECK:             %[[OBJECT_FIFO_1:.+]] = amdaie.logicalobjectfifo.from_memref
// CHECK:             amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_1]] [] [] [] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
// CHECK:             amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
// CHECK:             %[[TOKEN_0:.+]] = amdaie.npu.half_dma_cpy_nd async %[[CONNECTION]](%[[OBJECT_FIFO_1]] [0, 0, 0, 0] [2, 4, 16, 16] [0, 64, 8, 1] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
// CHECK:             amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
// CHECK:             amdaie.npu.dma_wait(%[[TOKEN_0]] : !amdaie.async_token)
// CHECK:             %[[TOKEN_1:.+]] = amdaie.npu.half_dma_cpy_nd async %[[CONNECTION]](%[[OBJECT_FIFO_1]] [0, 0, 0, 32] [2, 4, 16, 16] [128, 64, 8, 1] bd_id = %[[BD_ID]] channel = %[[CHANNEL_0]]) : !amdaie.logicalobjectfifo<memref<2048xi32>>
// CHECK:              amdaie.npu.half_dma_cpy_nd %[[CONNECTION]](%[[OBJECT_FIFO_0]] [] [] [] channel = %[[CHANNEL_1]]) : !amdaie.logicalobjectfifo<memref<2048xi32, 1 : i32>, 2>
// CHECK:              amdaie.npu.dma_wait(%[[TOKEN_1]] : !amdaie.async_token)
```
This is a style suggestion, but I think using indentation is more readable. Although we're not exactly consistent on this format, most tests use this indented format.

Reply:
thanks! done